lib/build.js

```diff
@@ -28,18 +34,32 @@ const build = async ({ dir, verbose, skipNextBuild }) => {
 
   await init({ dir, verbose })
 
-  logv({ skipNextBuild })
-  if (!skipNextBuild) {
-    const nextBuild =
-      require(`${packageDirWithNext}/node_modules/next/dist/build/index`).default
+  logv({ skipBuild, onlyApiFiles })
 
-    await nextBuild(dir, null, false, false, false)
+  if (!skipBuild) {
+    const nextjsOptions = {
+      build: require(`${packageDirWithNext}/node_modules/next/dist/build/index`).default,
+      generateRoutes: require(`${dir}/.nsm/scripts/generate-routes.js`).generateRoutes,
+    }
+
+    const nextlessOptions = {
+      build: require(`${dir}/.nsm/scripts/nsm-build.js`).nsmBuild,
+      generateRoutes: require(`${dir}/.nsm/scripts/generate-routes.js`).generateNsmRoutes, 
+    }
```

Comment:
Since `require` is i/o that might fail, we should conditionally execute this. We can move it to a function with early return or use if/else.

Reply:
what if we have something like:
```suggestion
    const getNextjsOptions = () =>  ({
      build: require(`${packageDirWithNext}/node_modules/next/dist/build/index`).default,
      generateRoutes: require(`${dir}/.nsm/scripts/generate-routes.js`).generateRoutes,
    })

    const getNextlessOptions = () => ({
      build: require(`${dir}/.nsm/scripts/nsm-build.js`).nsmBuild,
      generateRoutes: require(`${dir}/.nsm/scripts/generate-routes.js`).generateNsmRoutes, 
    })
```